web/src/components/layout/Panel.vue

```diff
@@ -1,23 +1,63 @@
 <template>
   <div class="rounded-md w-full shadow overflow-hidden bg-gray-300 dark:bg-dark-gray-700">
-    <div v-if="title" class="font-bold text-gray-200 bg-gray-400 dark:bg-dark-gray-800 px-4 py-2">{{ title }}</div>
-    <div class="w-full p-4 bg-white dark:bg-dark-gray-700 text-color">
-      <slot />
+    <component
+      :is="collapsable ? 'button' : 'div'"
+      v-if="title"
+      type="button"
+      class="flex w-full font-bold gap-2 text-gray-200 bg-gray-400 dark:bg-dark-gray-800 px-4 py-2"
+      @click="collapsed = !collapsed"
+    >
+      <Icon
+        v-if="collapsable"
+        name="chevron-right"
+        class="transition-transform duration-150 min-w-6 h-6"
+        :class="{ 'transform rotate-90': !isCollapsable }"
+      />
+      {{ title }}
+    </component>
+    <div
+      :class="{
+        'max-h-screen': !isCollapsable,
+        'max-h-0': isCollapsable,
+      }"
+      class="transition-height duration-150 overflow-hidden"
+    >
+      <div class="w-full p-4 bg-white dark:bg-dark-gray-700 text-color">
+        <slot />
+      </div>
     </div>
   </div>
 </template>
 
 <script lang="ts">
-import { defineComponent } from 'vue';
+import { computed, defineComponent, ref } from 'vue';
+
+import Icon from '~/components/atomic/Icon.vue';
 
 export default defineComponent({
   name: 'Panel',
+  components: { Icon },
 
   props: {
     title: {
       type: String,
       default: '',
     },
+
+    collapsable: {
+      type: Boolean,
+    },
+  },
+
+  setup(props) {
+    const collapsed = ref(false);
+
+    const isCollapsable = computed(() => props.collapsable && collapsed.value);
```

Comment:
I think `isCollapsable` isn't the best name here. As I don't have a better name as well, we could maybe just use `_` for one of them:

```suggestion
    const _collapsed = ref(false);
    const collapsable = toRef(props, 'collapsable');
    const collapsed = computed(() => collapsable.value && _collapsed.value);
```